ruby/test/Issue_2560_Test.rb

```diff
@@ -0,0 +1,168 @@
+#  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+########################################################################################################################
+require 'openstudio'
+require 'logger'
+require 'openstudio/measure/ShowRunnerOutput'
+require 'fileutils'
+require 'minitest/autorun'
+require 'csv'
+
+class Issue_2560_Test < Minitest::Test
+    def test_Issue_2560
+        model = OpenStudio::Model::Model.new
+        story2 = OpenStudio::Model::BuildingStory.new(model)
+        
+        perimeter = OpenStudio::Point3dVector.new 
+        perimeter.push(OpenStudio::Point3d.new(4, 2, 3))
+        perimeter.push(OpenStudio::Point3d.new(4, 6, 3))
+        perimeter.push(OpenStudio::Point3d.new(18, 6, 3))
+        perimeter.push(OpenStudio::Point3d.new(18, 2, 3))
+        #OptionalSpace
+        sp1 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp1)
+        sp1 = sp1.get
+        sp1.setBuildingStory(story2)
+        tz1 = OpenStudio::Model::ThermalZone.new(model)
+        sp1.setThermalZone(tz1)
+        
+        story1 = OpenStudio::Model::BuildingStory.new(model)
+
+        perimeter.clear()
+        perimeter.push(OpenStudio::Point3d.new(0, 0, 0))
+        perimeter.push(OpenStudio::Point3d.new(0, 3, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 3, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 0, 0))
+        #OptionalSpace
+        sp2 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp2)
+        sp2 = sp2.get
+        sp2.setBuildingStory(story1)
+        tz2 = OpenStudio::Model::ThermalZone.new(model)
+        sp2.setThermalZone(tz2)
+
+        perimeter.clear()
+        perimeter.push(OpenStudio::Point3d.new(0, 3, 0))
+        perimeter.push(OpenStudio::Point3d.new(0, 5, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 5, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 3, 0))
+        #OptionalSpace
+        sp3 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp3)
+        sp3 = sp3.get
+        sp3.setBuildingStory(story1)
+        tz3 = OpenStudio::Model::ThermalZone.new(model)
+        sp3.setThermalZone(tz3)
+
+        perimeter.clear()
+        perimeter.push(OpenStudio::Point3d.new(0, 5, 0))
+        perimeter.push(OpenStudio::Point3d.new(0, 8, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 8, 0))
+        perimeter.push(OpenStudio::Point3d.new(22, 5, 0))
+        #OptionalSpace
+        sp4 = OpenStudio::Model::Space::fromFloorPrint(perimeter,3,model)
+        assert(sp4)
+        sp4 = sp4.get
+        sp4.setBuildingStory(story1)
+        tz4 = OpenStudio::Model::ThermalZone.new(model)
+        sp4.setThermalZone(tz4)
+
+        # Model before intersection
+        outpath = File.join(File.expand_path('../../.') + "/build/resources/model/","2560_before.osm") 
```

Comment:
I think the "build" directory needs to be configured. I think this is based on an assumption about where the build directory is located, which cannot be guaranteed. @jmarrec do you think you could help @ggartside configure this?

Reply:
The main point here is that these are **ruby** tests instead of being **Google Test**s (C++). Google Tests would have access to resourcesPath https://github.com/NREL/OpenStudio/blob/a007b7801e3bb05731f012f585facdd814ef31fb/resources/resources.hxx.in#L6-L9

I'm not sure why these are ruby tests, whether that's written in the scope of work or not. I also realize it would take a bit of work to make them C++ tests, so I don't really have an opinion. I can propose a safer workaround for now.

Reply:
I removed all save code from the ruby tests in fb25ab9186fbd6fb71ed041858e18380bd436ca2

Reply:
The questiona bout whether we should change these ruby tests to be C++ tests still stands. (Also, I think there are a couple of tests that are both in C++ and ruby tests)

Reply:
We are contracted to provide both c++ and ruby tests so we started with a set of c++ tests then duplicated them in ruby at the end of the project, There are some tests we haven't pushed yet because we were trying to find out how to specify the location of model files but now that might not be an issue any more if some of these tests are not require or we create the model through the API